base/math.jl

```diff
@@ -715,6 +715,13 @@ julia> .√(1:4)
 """
 sqrt(x)
 
+"""
+    fourthroot(x)
+
+Return the fourth root of `x` by applying `sqrt` twice successively.
+"""
+fourthroot(x::Union{Float32,Float64}) = sqrt(sqrt(x))
```

Comment:
this should have a much less strict type restriction (probably `Number`?)

Reply:
That was my first try but I got some errors about methods not being found so I assumed I was doing it wrong, which led me to change to my current code. These are the errors I get:

```julia
julia> methods(∜)
# 5 methods for generic function "fourthroot" from Base.Math:
 [1] fourthroot(a::Float16)
     @ math.jl:1549
 [2] fourthroot(x::Real)
     @ math.jl:1566
 [3] fourthroot(a::ComplexF16)
     @ math.jl:1550
 [4] fourthroot(x::Number)
     @ math.jl:723
 [5] fourthroot(::Missing)
     @ math.jl:1571

julia> ∜1
ERROR: MethodError: no method matching fourthroot(::Float64)

Closest candidates are:
  fourthroot(::Real)
   @ Base math.jl:1566
  fourthroot(::Number)
   @ Base math.jl:723
  fourthroot(::Float16)
   @ Base math.jl:1549
  ...

Stacktrace:
 [1] fourthroot(x::Float64)
   @ Base.Math ./math.jl:1568
 [2] fourthroot(x::Int64)
   @ Base.Math ./math.jl:1569
 [3] top-level scope
   @ REPL[1]:1
```

Which doesn't really make sense to me because `methods()` clearly lists `fourthroot(x::Number)`. So I'm not sure `∜1` isn't calling the `Number` method. This also happens `∜1.0`, `∜1f0`, etc.

I must be missing something obvious here...

Reply:
Also, defining the same thing from the REPL works fine:

```julia
julia> fourthroot(x::Number) = sqrt(sqrt(x))
fourthroot (generic function with 1 method)

julia> fourthroot(1.0)
1.0
```

Reply:
I think the problem was probably a result of stacking methods in a repl. I think that if you just relax the type restriction and rebuild it will all work out.

Reply:
The results below are done right after a clean compile (`make clean ; make ; ./julia`):

```julia
julia> fourthroot(1.0f0)
ERROR: UndefVarError: `fourthroot` not defined
Stacktrace:
 [1] top-level scope
   @ REPL[1]:1

julia> ∜(1.0f0)
ERROR: MethodError: no method matching fourthroot(::Float32)

Closest candidates are:
  fourthroot(::Real)
   @ Base math.jl:1566
  fourthroot(::Number)
   @ Base math.jl:723
  fourthroot(::Float16)
   @ Base math.jl:1549
  ...

Stacktrace:
 [1] fourthroot(x::Float32)
   @ Base.Math ./math.jl:1568
 [2] top-level scope
   @ REPL[1]:1
```

these are the first command on the REPL. So I think I must be missing something, no?

Reply:
can you commit the change so I can look at it?

Reply:
For sure! Thanks

Reply:
yeah I'm pretty sure the problem is the `::Real` method you've added that breaks the `Number` one.